### DISCUSSION TITLE
Meeting Notes: OpenDCS REST API CWMS Implementation Kickoff

### DISCUSSION THREAD

**Discussion by adamkorynta** [opendcs/rest_api#211] [Announcements]
RMA is starting in on work for the CWMS implementation.

attendees:

- Fauwaz Hanbali (HEC)
- Mike Neilson (HEC)
- Karl Tarbet (HEC)
- Oskar Hurst (HEC)
- Peter Morris (RMA)
- William Jonassen (OLAV)
- Adam Korynta (RMA)
- Art Armour (NWP)

Notes:

- The original work plan targeted refactoring existing DAO classes to split up interfacing between CWMS and OpenTSDB implementations (taking into account other future implementations)
- Instead RMA is refocusing on using existing OpenDCS repo artifact db interfacing and pulling out all SQL in REST API to use common SQL. In conjunction the service loading of the database implementations will utilize schema updates that define implementation configuration. https://github.com/opendcs/opendcs/pull/520
- DAO integration testing handled in OpenDCS main codebase SQL 
- REST API integration testing and unit testing of data model/DTO classes will occur in the REST API. The CWMS flyway migration will help in this effort: https://github.com/opendcs/opendcs/pull/744
- The jetty distribution will be deprecated in favor of a WARfile only distribution which can be utilized in the Cloud via a jetty docker container.
- RMA will revise the existing workplan to address the above and put into the GitHub Wiki page (will add a reference link here).

**Comment by MikeNeilson** [comment]
FYI https://github.com/opendcs/opendcs/wiki/Project-%E2%80%90-Merging-the-DECODES-Database-and-TimeseriesDb-classes was brought up during the meeting, I've updated it to match the current PR, [#681](https://github.com/opendcs/opendcs/pull/681), which replaces the initial effort of https://github.com/opendcs/opendcs/pull/520.

I've closed #520, left it open for reference but clearly that caused some confusion.

also not discussed was https://github.com/opendcs/opendcs/pull/716. After that is merged or just set as the default, we'll next work on automating releases so any authorized party can queue up a release or release candidate which will be needed for the REST API to depend on the coming changes.
